webrender/src/resource_cache.rs

```diff
@@ -116,6 +116,40 @@ struct ImageResource {
     dirty_rect: Option<DeviceUintRect>
 }
 
+struct ImageTemplates {
```

Comment:
The reason is the borrow checker. Without the separate struct the resource cache ends up being "entirely" borrowed when passed to the blob renderer which can't happen because we are already borrowing it partially (the blob renderer itself).